### ISSUE TITLE
[eFEAT] Training week preview in Training Tabs

### ISSUE THREAD

**Issue by mistermax80** [akasolace/HO#234]
Continuing the discussion started on #222 
Insert in training tabs the information on training on week (icon and tooltip).
The final effect it will be as following:
![image](https://user-images.githubusercontent.com/15519907/65895415-b8bfc880-e3ab-11e9-9098-13794ba8f195.png)

In training tab (asis function) the colors are 2:
- green for player in position that receive all training
- yellow for player in position that receive partial training
for example in case of cross traiinig:
- green for wings
- yellow for back wings
this logics consider only position and future training and these don't see the lineup.

The yaute logic do:
green (ligth/or no light): for player received total training in terms of time
grey: for player will receive training in terms of time.

I would like see in the tab training both information. My idea is use yaute logic for to show the icon and tooltip, and leave current logic as is.

The yaute logics is an static function, and it is usable in each part of code.

@yaute74 , sorry, I would like ask you, explain to @akasolace that I say. Could you help me?

**Comment by yaute74** [comment]
In principle I agree. I think you can use the same object than me for the user column and leave the rest as it is.

Just a remark, aren't you afraid it'll give too much information to the user ?

**Comment by yaute74** [comment]
@mistermax80  you must use SpielerLabelEntry object

**Comment by mistermax80** [comment]
@yaute, yes... Respect to too much information to user, if you think this I can think to insert a preference for this.
I can say I will use the all information. I will have these information in only screen it is comfortable!

**Comment by yaute74** [comment]
@mistermax80  ok great :)

**Comment by mistermax80** [comment]
Can it useful insert a new column for to show the only skill value that receive train?

**Comment by akasolace** [comment]
>Can it useful insert a new column for to show the only skill value that receive train?
the user knows which skill he is training so I am not sure this will be very useful. Addiitonaly the tab is already quite packed

**Comment by mistermax80** [comment]
Ok I agree!

**Comment by mistermax80** [comment]
@yaute74 , I tried to do it, but I have some difficult.
When you can, can you help me on this? We will for version 3, so nothing hurry.

**Comment by mistermax80** [comment]
I don't able to set the render cell in order to build a cell name with training icon and player name.
So when I update the table rendering, in the firts cell appear the java object name and not icon+player name.

**Comment by yaute74** [comment]
@mistermax80 ok I will check when I have the time...

**Comment by yaute74** [comment]
@mistermax80  Quickly you must modify the renderer in module.training.ui.renderer.OutputTableRenderer method getTableCellRendererComponent for the column 0

**Comment by mistermax80** [comment]
@yaute74 , thanks so much, but I don't belive it is, My idea is build an object composed by training icon and jlabel for name, but the object doesn't printed... I don't undestand how cell name of the overview works....

**Comment by mistermax80** [comment]
@yaute74 , you said rigth...in that method, inserting following, begin works:
        if (column == 0) {
            JLabel jl = (JLabel) value;
            return jl;
        }

The way is correct.

Thanks!

**Comment by mistermax80** [comment]
@yaute74, now the icon is visible and the tooltip is ok, but I don't able to change backgroung color when the row is selected. You can see the file src/main/java/module/training/ui/renderer/OutputTableRenderer.java , line 31 to 35.
Could you help me?

![Schermata 2019-10-20 alle 00 45 02](https://user-images.githubusercontent.com/15519907/67152284-196b6280-f2d3-11e9-9863-86733e2ae9ed.png)

**Comment by yaute74** [comment]
@mistermax80 sorry for the delay, I was not available this week-end.
In the renderer you must do something like:
```Java

if (column == 0) {
    JLabel jl = (JLabel) value;
    if (isSelected)
        jl.setBackground(SELECTION_BG);
    else
        jl.setBackground(Color.WHITE);
    return jl;
}
```

**Comment by mistermax80** [comment]
I made but doesn't work as expected...
Could you see the code in my branch?

**Comment by yaute74** [comment]
It's because your jlabel is transparent.
So in module.training.ui.comp.PlayerNameCell constructor add

```Java

this.setOpaque(true);

**Comment by mistermax80** [comment]
thanks so much, now it works...

**Comment by mistermax80** [comment]
in time miss, I work on this, now, my proposal is:
![image](https://user-images.githubusercontent.com/15519907/68630379-6f54b400-04e7-11ea-90dc-5b51c9f43060.png)
I remember you the old bar style is very bad:
![image](https://user-images.githubusercontent.com/15519907/68630417-99a67180-04e7-11ea-9c54-319a12035723.png)

I would like to know what you think, if you want I will do configurable in preference.

**Comment by akasolace** [comment]
@mistermax80  so before the bar was only the integer value. Now you add a light color for the decimal part.   But what is the last bar, the one related to training ?

**Comment by mistermax80** [comment]
@akasolace  lthe old bar has the skillvalue with decimals, and for the skill trained, the bar is of the value of the end the training. If you see in my previous post (old) you can see the passing bar is very long, this because I train passing and the long bar represents the value of end training.
I made the following modify respect to old bar:
- add form
- reorder as HT
- colour behaviour for form and stamina as HT
- the skill bar in green as HT and different color 
summarize, I didn't modify the beaviuos for decimal and different color for future trining.
I'm going to add a tooltip that shows the value for example: 5 / .67 / 10.
Other thing I like to add an way to set the target train, but I should to think on this.

**Comment by akasolace** [comment]
@mistermax80  ok I see 
maybe we can simply indicate plain / light for integer/decimal value and use a different color for the currently trained skill. might be less confusing.    @DavidatorusF @yaute74  any remarks on this ? Do you like proposed implementation?

**Comment by yaute74** [comment]
Sorry I was busy this week.
For me it's ok

**Comment by mistermax80** [comment]
@akasolace @yaute74 
I asked to @DavidatorusF an his opinion on this, him prefer blue insted green and the decimal skill bar is too little.

So, I would like to ahead in this way:
- try color bar to blue
- remove the little decimal skill bar
- try to insert an highlithg for give evidence on the skill trained

**Comment by DavidatorusF** [comment]
was a joke about the blue ^^
green is nice

about the decimal skill, it's only cause of my glasses, I need new one..
when decimal skill become bigger (than on our your screenshot), I think we can see it better

**Comment by mistermax80** [comment]
I get it your joke, but I would like to consider an different color respect to HT.
the high value for decimal it will be 0,999 and the effect in the bar is an very small bar, and so I am not convince the its efficacity.
I will a try and after we will decide.

**Comment by DavidatorusF** [comment]
may be use green more different...
![image](https://user-images.githubusercontent.com/45043004/69335105-73f23880-0c5c-11ea-90c7-6642da9e5a13.png)

**Comment by mistermax80** [comment]
ok I made an a try in both method

**Comment by mistermax80** [comment]
Other test:
![image](https://user-images.githubusercontent.com/15519907/69441732-86976b00-0d4b-11ea-8375-5af66e84fd84.png)

**Comment by mistermax80** [comment]
![image](https://user-images.githubusercontent.com/15519907/69442648-3de0b180-0d4d-11ea-8f1f-28767c62578f.png)

@akasolace @yaute74 @DavidatorusF what do you think?

**Comment by DavidatorusF** [comment]
ah ah ah, I like thee blue one

but green is the best for HO

**Comment by akasolace** [comment]
I am ok with both, I will let you decide

**Comment by mistermax80** [comment]
Blue Is nice, but green is more correct related to HO colours.
@akasolace @yaute74 @DavidatorusF 
Do you want I manage some preferences for:
1 - to show training preview
2 - to show new colour green bar.

Let me know what you want.

**Comment by akasolace** [comment]
> Preference to show training preview 

You man to activate/deactivate the functionality? In that case I would say no

> Preference to choose color bar

 clearly no

**Comment by mistermax80** [comment]
Yes I mean that.
Ok new for all!

**Comment by mistermax80** [comment]
@akasolace 
By 5acfa63eb6690062f56f014d237cf7f22df0ffa5, the following situation, it can be consider stable and final.
I wait release HO 2.0 for send the relative PR (after rebase or merge... see slack).

![image](https://user-images.githubusercontent.com/15519907/69819968-4c711200-1200-11ea-9dff-8cdb9c84a39f.png)

**Comment by akasolace** [comment]
@mistermax80 ok great thank you. Stable will be release this weekend.

**Comment by mistermax80** [comment]
In order to check and programme the training or other players details, I'm going to insert in training tab the note panel (present in player details of the plyaer overview tab)

**Comment by akasolace** [comment]
> In order to check and programme the training or other players details, I'm going to insert in training tab the note panel (present in player details of the plyaer overview tab)

did not get it

**Comment by mistermax80** [comment]
![image](https://user-images.githubusercontent.com/15519907/69878911-6b39dc00-12c6-11ea-9370-8f86a81d3583.png)

![image](https://user-images.githubusercontent.com/15519907/69878987-9b817a80-12c6-11ea-8f4e-8849823bc486.png)

**Comment by mistermax80** [comment]
By this: 2f0203c0c7daeb5e2c3cac9c27d3624eb2fd15f9
The final version, I promises... the real final... 😄 
![image](https://user-images.githubusercontent.com/15519907/69883360-18ffb780-12d4-11ea-9a3e-b501bb4baaf7.png)

**Comment by akasolace** [comment]
@mistermax80 what is it you put as note?

**Comment by mistermax80** [comment]
free text "lorem ipusum" inserted for test, for default is blank else it is the same player note in player overview tab

**Comment by akasolace** [comment]
@mistermax80 for my info because I never used that: the player note is not editable, it is the one from HT, right ?
So it is synced at download time, right?

**Comment by mistermax80** [comment]
@akasolace no, it is that of HO, in player overview:
![image](https://user-images.githubusercontent.com/15519907/69887301-887ea280-12e6-11ea-9d60-efd31226d969.png)

HT notes are available for only supporter, but it can be an new fetaures --> intresting for user uses HT notes

**Comment by mistermax80** [comment]
PR ready #300

**Comment by akasolace** [comment]
closed by f9f6b8e
